tony-azkaban/src/test/java/com/linkedin/tony/azkaban/TestTonyJob.java

```diff
@@ -63,6 +70,48 @@ public String getWorkingDirectory() {
     Assert.assertTrue(args.contains(TonyJobArg.SHELL_ENV.tonyParamName + " E1=e1"));
   }
 
+  /**
+   * Check if the flow level information is passed to the tony job through configuration.
+   */
+  @Test
+  public void testFlowInfoPropagation() {
+    final Props jobProps = new Props();
+    jobProps.put(TonyJobArg.HDFS_CLASSPATH.azPropName, "hdfs://nn:8020");
+    jobProps.put(CommonJobProperties.PROJECT_NAME, "unit_test");
+    jobProps.put(CommonJobProperties.FLOW_ID, "1");
+    jobProps.put(CommonJobProperties.EXEC_ID, "0");
+
+    final TonyJob tonyJob = new TonyJob("test_tony_job", new Props(), jobProps, log) {
+      @Override
+      public String getWorkingDirectory() {
+        return System.getProperty("java.io.tmpdir");
+      }
+    };
+    tonyJob.getMainArguments();
+
+    org.apache.hadoop.conf.Configuration conf =
+        new org.apache.hadoop.conf.Configuration(false);
+    conf.addResource(
+        new Path(tonyJob.getWorkingDirectory() + "_tony-conf-test_tony_job/tony.xml"));
```

Comment:
I think it's not good that `TonyJob.getMainArguments()` has the side effect of writing out this XML file. Can you move the XML file writing logic out into a separate method that's called from the constructor?

BTW, I don't think it's necessary to write a test verifying that the conf was written to the `tony.xml`; I think it's enough to verify that the application tags are added to the `Configuration` object. (You might need to add a method to `TonyJob` that exposes the `Configuration` for testing purposes.)

Reply:
That's a good point. I split the processing Conf and writing it to xml file into two separate methods,  getJobConfiguration() and  setupJobConfigurationFile(). The former is called from the TonyJob constructor, the latter does IO, which I don't think is suitable to put in the constructor. Instead, it is called in run() method as a prep step as setupHadoopOpts().